Backend/Data/PersonRepository.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Backend.Data.Abstraction;
+using Backend.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backend.Data
+{
+    internal class PersonRepository: IPersonRepository
+    {
+        private MyDbContext _context;
+        public PersonRepository(MyDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IList<Person>> GetAllStars()
+        {
+            var personIds =  await _context.Stars.Select(x=>x.PersonId).ToListAsync();
+            return await _context.People.Where(x=>personIds.Contains(x.Id)).ToListAsync();
+        }
+
+        public async Task<IList<Person>> GetAllDirectors()
+        {
+            var personIds =  await _context.Directors.Select(x=>x.PersonId).ToListAsync();
+            return await _context.People.Where(x=>personIds.Contains(x.Id)).ToListAsync();
+        }
+
+        public async Task<Person> GetStarOrDirectorById(long id)
```

Comment:
Related to my comment above  this is GetStarOrDirector for personId I guess? Not movie Id that is why I think we can make some changes for getting a director for movieId (or maybe it already exists in the movie repostiry I have to check)